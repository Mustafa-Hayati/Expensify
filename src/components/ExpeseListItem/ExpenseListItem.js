import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({
  expense: { description, amount, createdAt, id },
}) => {
  return (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>
        ${amount / 100} - {createdAt}
      </p>
    </div>
  );
};

export default ExpenseListItem;
