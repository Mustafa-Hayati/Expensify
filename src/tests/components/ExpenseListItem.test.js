import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "./../../components/ExpeseListItem/ExpenseListItem";
import expenses from "./../fixtures/expenses";

test("should render ExpenseListItem with a given expense", () => {
  const wrapper = shallow(
    <ExpenseListItem key={expenses[0].id} expense={expenses[0]} />
  );
  expect(wrapper).toMatchSnapshot();
});
