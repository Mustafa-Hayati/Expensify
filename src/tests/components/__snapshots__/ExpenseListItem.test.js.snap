// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with a given expense 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $
    1.95
     - 
    0
  </p>
</div>
`;
